### PR TITLE
[Add] SparseMLEvalRegistry

### PR DESCRIPTION
**Pull request by rahul-tuli**
# Registry

This PR adds a `SparseMLEvalRegistry` for registering evaluation integrations.

## Design

registry

Note: during collection we only use the integration name to figure out the integration

## Test Plan

Automated Tests

Based off of #1992 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206432804121697

### CODE REVIEW

File: src/sparseml/evaluation/registry.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    # The integration locations must be hardcoded here to be collected
+    # this is a time being solution until we move to a config based
+    # solution
+
+    _LOGGER.debug(
+        f"Auto collection of {name} integration for eval failed. "
+        "The integration must be registered and collected/imported "
+        "manually."
+    )
```
Comment (Satrat) [inline src/sparseml/evaluation/registry.py:71]: A bit confused whats going on here, why does the automatic collection fail?

File: src/sparseml/evaluation/registry.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        """
+        Resolve an evaluation integration by name.
+
+        :param name: The name of the evaluation integration to resolve
+        :param args: The arguments to pass to the evaluation integration
+        :param kwargs: The keyword arguments to pass to the evaluation integration
+        :return: The evaluation integration associated with the name
+        """
+
+        collect_integrations()
```
Comment (rahul-tuli) [inline src/sparseml/evaluation/registry.py:45]: name should be passed here!

File: src/sparseml/evaluation/registry.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    This is done by importing the module associated with each integration.
+    This function is called automatically when the registry is accessed.
+    The specific integration(s) `callable` must be decorated with the
+    `@SparseMLEvaluationRegistry.register` decorator.
+
+    :param name: The name of the integration to collect, is case insentitive
+    """
+
+    # The integration locations must be hardcoded here to be collected
```
Comment (bfineran) [inline src/sparseml/evaluation/registry.py:63]: maybe we could move this to a json/config file in the repo and parse it here

**Review by bfineran — APPROVED** [review]

File: src/sparseml/evaluation/registry.py (hunk 3 above)
Comment (rahul-tuli) [reply src/sparseml/evaluation/registry.py]: Done!

**Pull request merged**